### PR TITLE
Update GitHub actions workflows

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
**Changes**
- Use Ubuntu 22.04
- Use Java 17
- Use Java Temurin distribution
- Use official Gradle setup action
- Use consistent naming scheme
- Use organization check instead of repository check

**Issues**

### CODE REVIEW

File: .github/workflows/repo-stale.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: Issue stale check
+name: Repo Stale
```
Comment (Maxr1998) [inline .github/workflows/repo-stale.yaml:1]: Not a fan of the "Repo ..." names. Would be prefer something more descriptive like "Stale Issues Check" or "Check for stale issues".

File: .github/workflows/repo-merge-conflict.yaml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-name: Merge conflict labeler
+name: Repo Merge conflict
```
Comment (Maxr1998) [inline .github/workflows/repo-merge-conflict.yaml:1]: Same here, "Label Merge Conflicts" would work imo.

File: .github/workflows/repo-stale.yaml (hunk 1 above)
Comment (Maxr1998) [reply .github/workflows/repo-stale.yaml:1]: Capitalisation consistency on the titles is a bit weird too. If titles should be uppercase, all words should be.
We could theoretically follow the MusicBrainz English styleguide for titles ^^

Comment (nielsvanvelzen) [reply .github/workflows/repo-stale.yaml:1]: For consistency everything repository related changed to start with "Repo" so we now have App, Repo and Gradle

Comment (Maxr1998) [reply .github/workflows/repo-stale.yaml:1]: For the files that's ok, but the title shown in GitHub is a downgrade imo. It's less easy to understand what's happening.

**Review by Maxr1998 — APPROVED** [review]

**Pull request merged**